drivers: ptp_clock: add ptp_clock shell commands

yangbolu1991 opened the pull request:
Added ptp_clock shell commands to check, use, and verify ptp clock. Supported get, set, adjust, rate adjust, and selftest functions.

yangbolu1991 committed:
drivers: ptp_clock: add ptp_clock shell commands

Added ptp_clock shell commands to check, use, and verify
ptp clock. Supported get, set, adjust, rate adjust, and
selftest functions.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>

yangbolu1991 commented on drivers/ptp_clock/ptp_clock_shell.c:
```diff
@@ -0,0 +1,239 @@
+	}
+
+	shell_print(sh, "adjust 3 seconds: %"PRIu64".%09u", tm.second, tm.nanosecond);
+
```
Okay, let me give a new version to review.
Will give a better description in help and print messages to explain what the selftest does.
Actually it's very easy to understand the tests.
Thanks.

yangbolu1991 commented:
Updated to v2. Changes include,
- Improved code per @jukkar suggestion.
- Made user to configure selftest args.

Thanks.

awojasinski commented on drivers/ptp_clock/ptp_clock_shell.c:
```diff
@@ -0,0 +1,253 @@
+	}
+
+	/* set 'time' seconds and read back to verify clock setting/getting */
+	tm.second = seconds;
```
`seconds` should be `uint64_t`. Please change `shell_strtol()` to `shell_strtoull()`

yangbolu1991 commented:
Updated to v3. Changes include,
- Fixed set time seconds with uint64_t.

Thanks.

yangbolu1991 commented:
Updated to v4. Changes include,
- Just found I making typo in v3 changes. Should have used `shell_strtoull()` not `shell_strtoll()`. Fixed.

Thanks.

yangbolu1991 commented:
It seems some CI issue on code base.
Rebased, thanks.

jukkar reviewed (approved):
LGTM

yangbolu1991 commented:
@awojasinski @tbursztyka 
Could you help to review the new version?
Thank you very much.

tbursztyka reviewed (approved):
following other reviewers, so +1  to unlock this. (hopefully a more relevant ptp maintainer will arrive some day)
